main/acle.md

```diff
@@ -1888,6 +1889,146 @@ or compiler-generated functions.
+
+1. when applied to a function it becomes one of the versions. Function with the same
+name may exist with multiple versions in the same translation unit.
+2. One `"default"` version of the function is required to be provided.
+   a. Implicitly, without this attribute,
+   b. or explicitly providing the `"default"` in the attribute.
+4. All instances of the versions shall share the same function signature.
+
+The attribute `__attribute__((target_clones("name",...)))` expresses the following:
+
+1. when applied to a function the compiler emits multiple versions based on the
+arguments.
+   a. One of them is implicitly the `"default"`.
+   b. If the `"default"` matches with an other explicitly provided version the
+compiler can emit only one function instead of the two.
+2. If a name is not recognized the compiler should ignore it.
+
+`__ARM_FEATURE_FUNCTION_MULTI_VERSIONING` is defined to 1 if the versioning
+mechanism described in this section is available in the compiler and it is enabled.
+
+### Name mangling
+
+The `"default"` version is not mangled top of the languge specific name mangling.
+
+Function version to be mangled for C/C++. `<vendor-specific suffix>` should hold the version
+information [[cxxabi]](#cxxabi) as `_(M<name>)*` `name` is from tables below.
+
+If multiple features are requested then those shall be appended in priority order.
+
+For example:
+``` c
+__attribute__((target_clones("crc32", "aes+sha1")))
+int foo(){..}
+```
+will produce these mangled names for C languae: `foo`, `foo._Mcrc32`, `foo._Msha1Maes`.
+
+
+### Mapping
+
+The following table lists the architectures feature mapping for AArch32
+
+   | **Priority** | **Architecture name**            | **Name**        | **Dependent feature registers** |
+   | ------------ | -------------------------------- | --------------- | ------------------------------- |
+   | 0            | N/A                              | default         | ```N/A```                       |
+   | 90           | CRC32 instructions               | crc32           | ```ID_ISAR5.CRC32 == 0b0001```  |
+   | 100          | SHA1 instructions                | sha1            | ```ID_ISAR5.SHA1 == 0b0001```   |
+   | 110          | SHA2 instructions                | sha256          | ```ID_ISAR5.SHA2 == 0b0001```   |
+   | 120          | AES instructions                 | aes             | ```ID_ISAR5.AES >= 0b0001```    |
+   | 130          | VMULL (polynomial) instructions  | vmull           | ```ID_ISAR5.AES == 0b0002```    |
+
+The following table lists the architectures feature mapping for AArch64
+
+   | **Priority**  | **Architecture name**    | **Name**      | **Dependent feature registers**           |
+   | ------------- | ------------------------ | ------------- | ----------------------------------------- |
+   | 0             | N/A                      | default       | N/A                                       |
```

Comment:
Sometimes you write N/A and sometimes you write `N/A`. I think we can stick with the former (no ticks).